src/page-manager.js

```diff
@@ -163,15 +174,19 @@ define([
         },
         /**
          * Only pushes a route to the history API but doesn't navigate to it
+         * @param {String|Object} path url or definition object
          */
         pushRoute : function(path) {
-            this._navigate(path, { trigger : false });
+            var route = ('string' === typeof path) ? path : this.buildPath(path);
+            this._navigate(route, { trigger : false });
         },
         /**
          * Replace the current History API entry
+         * @param {String|Object} path url or definition object
          */
         replaceCurrentRoute : function(path) {
-            this._navigate(path, { replace : true });
+            var route = ('string' === typeof path) ? path : this.buildPath(path);
```

Comment:
Separate methods: replaceCurrentRoute(route) and replaceCurrentRoute(route) // definition